### PR TITLE
Feature/v7 patchset rebase squash new anjo

### PR DESCRIPTION
**Pull request by AntonJohansson**

### CODE REVIEW

File: target/hexagon/idef-parser/parser-helpers.h [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+void yyerror(YYLTYPE *locp,
+             yyscan_t scanner __attribute__((unused)),
+             Context *c,
+             const char *s);
+
+#ifndef NDEBUG
+#define yyassert(context, locp, condition, msg)              \
+    if (!(condition)) {                                      \
+        yyerror(locp, (context)->scanner, (context), (msg)); \
+        abort();                                             \
```
Comment (aleclearmind) [inline target/hexagon/idef-parser/parser-helpers.h]: This makes us `assert` every time there's an error, don't we want to print them all and fail at the end?

File: target/hexagon/idef-parser/README.rst [hunk 2]
```diff
@@ -0,0 +1,714 @@
...
+
+   sudo unshare -p sudo -u <USER> bash -c \
+   'env -i <qemu-hexagon full path> -d cpu <TEST>'
+
+And do the same for your implementation, the generated execution traces will be
+inherently aligned and can be inspected for behavioral differences using the
+``diff`` tool.
+
+Example of debugging erroneous tinycode generator code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (aleclearmind) [inline target/hexagon/idef-parser/README.rst:510]: Very nice!

File: target/hexagon/idef-parser/parser-helpers.h [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+                    HexValue *op1,
+                    HexValue *op2);
+
+HexValue gen_cast_op(Context *c,
+                     YYLTYPE *locp,
+                     HexValue *src,
+                     unsigned target_width,
+                     HexSignedness signedness);
+
+/*
```
Comment (aleclearmind) [inline target/hexagon/idef-parser/parser-helpers.h]: Consider using doxygen (`/**`) everywhere you're adding these comments.

File: target/hexagon/genptr.c [hunk 4]
```diff
@@ -634,5 +648,150 @@ static void vec_to_qvec(size_t size, intptr_t dstoff, intptr_t srcoff)
...
+{
+    gen_sat_i64(dest, source, width);
+    TCGv_i64 ovfl_64 = tcg_temp_new_i64();
+    tcg_gen_setcond_i64(TCG_COND_NE, ovfl_64, dest, source);
+    tcg_gen_trunc_i64_tl(ovfl, ovfl_64);
+    tcg_temp_free_i64(ovfl_64);
+}
+
+/* Implements the fADDSAT64 macro in TCG */
+void gen_add_sat_i64(TCGv_i64 ret, TCGv_i64 a, TCGv_i64 b) {
```
Comment (aleclearmind) [inline target/hexagon/genptr.c:746]: Was this easy?

File: target/hexagon/gen_helper_funcs.py [hunk 5]
```diff
@@ -287,11 +287,14 @@ def main():
     hex_common.read_attribs_file(sys.argv[2])
     hex_common.read_overrides_file(sys.argv[3])
     hex_common.read_overrides_file(sys.argv[4])
+    if len(sys.argv) > 6:
```
Comment (aleclearmind) [inline target/hexagon/gen_helper_funcs.py:299]: If this condition is false it means that `idef-parser` is not enabled, is that right?

Comment (AntonJohansson) [reply target/hexagon/gen_helper_funcs.py:299]: Yes exactly, it's not the most transparent way of doing things tho. If we pass 5 arguments it's disabled, if we pass 6 it's enabled. Any suggestion for improving this?

Comment (aleclearmind) [reply target/hexagon/gen_helper_funcs.py:299]: Using `argparse`, but let's keep changes local for now. Maybe add a comment.

Comment (AntonJohansson) [reply target/hexagon/gen_helper_funcs.py:299]: Alright

File: target/hexagon/idef-parser/parser-helpers.h (hunk 1 above)
Comment (AntonJohansson) [reply target/hexagon/idef-parser/parser-helpers.h]: Nice catch, fixed!

**Pull request closed without merging**